frontends/p4/fromv1.0/programStructure.cpp

```diff
@@ -143,6 +145,13 @@ void ProgramStructure::createTypes() {
         auto type = it.first->type;
         createType(type, true, &converted);
     }
+    for (auto it : registers) {
+        if (it.first->layout) {
+            auto type = types.get(it.first->layout);
+            if (type->is<IR::Type_Header>())
+                createType(type, true, &converted);
```

Comment:
Register layouts should always be treated as structs, not headers (should not have valid bits), They're only declared as header_types in P4_14, because that is the only way to define structs in P4_14 -- whether something is a header or metadata is determined by the instance declaration.